src/mesh/coordinates.cxx

```diff
@@ -1260,6 +1274,10 @@ int Coordinates::geometry(bool recalculate_staggered,
     localmesh->recalculateStaggeredCoordinates();
   }
 
+  // Invalidate and recalculate cached variables
+  zlength_cache.reset();
+  zlength();
```

Comment:
Do we need to recalculate here? Not critical, but perhaps it's not needed in some models?

Reply:
Fair point. It is recalculated on demand, so we can drop it here. Will do that shortly!